libgadget/blackhole.c

```diff
@@ -585,28 +592,22 @@ blackhole_feedback_ngbiter(TreeWalkQueryBHFeedback * I,
     if(P[other].ID == I->ID) return;
 
      /* BH does not accrete wind */
-    if(P[other].Type == 0 && SPHP(other).DelayTime > 0) return;
+    if(P[other].Type == 0 && SPHP(other).DelayTime > 0)
+        return;
 
     struct SpinLocks * spin = BH_GET_PRIV(lv->tw)->spin;
 
     if(P[other].Swallowed && P[other].Type == 5)	/* we have a black hole merger */
     {
         if(BHP(other).SwallowID != I->ID) return;
 
-        lock_spinlock(other, spin);
-
         O->BH_CountProgs += BHP(other).CountProgs;
-
-        /* We do not know how to notify the tree of mass changes. so
-         * blindly enforce a mass conservation for now. */
+        /* Leave the swallowed BH mass around
+         * so we can work out mass at merger. */
         O->Mass += (P[other].Mass);
         O->BH_Mass += (BHP(other).Mass);
-        P[other].Mass = 0;
-        BHP(other).Mass = 0;
-
-        slots_mark_garbage(other, PartManager, SlotsManager);
```

Comment:
Do we no longer mark the swallowed particle a  garbage?

Reply:
That's right - we don't want to GC it as per #67 we want to keep the swallowed particle around for merger tree stuff.

Reply:
This is actually a much better way of tracking mergers than the usual merger tree code - I'm surprised other people haven't done it. It kind of obsoletes an entire complicated subfield...

Reply:
What is meant by obsoletes an entire subfield? Which subfield is that in reference to

Reply:
There are a whole bunch of people who build algorithms to follow the mergers of dark matter halos from incomplete temporal sampling of snapshots, for example https://arxiv.org/abs/1307.3577

Reply:
Ah okay interesting, thanks for the link